### PR TITLE
chore: update docstrings

### PR DESCRIPTION
**Pull request by masci**
Fixes https://github.com/deepset-ai/haystack/issues/7141

### CODE REVIEW

File: integrations/mistral/src/haystack_integrations/components/generators/mistral/chat/chat_generator.py [hunk 1]
```diff
@@ -20,9 +20,21 @@ class MistralChatGenerator(OpenAIChatGenerator):
...
 
+    Key Features and Compatibility:
+    - **Primary Compatibility**: Designed to work seamlessly with the Mistral API Chat Completion endpoint.
+    - **Streaming Support**: Supports streaming responses from the Mistral API Chat Completion endpoint.
+    - **Customizability**: Supports all parameters supported by the Mistral API Chat Completion endpoint.
+
+    This component uses the ChatMessage format for structuring both input and output,
+    ensuring coherent and contextually relevant responses in chat-based text generation scenarios.
+    Details on the ChatMessage format can be found in the
+    [OpenAI docs](https://github.com/openai/openai-python/blob/main/chatml.md)
```
Comment (anakin87) [inline integrations/mistral/src/haystack_integrations/components/generators/mistral/chat/chat_generator.py:31]: This URL is outdated (was previously deprecated by OpenAI).
I would refer to our docs: https://docs.haystack.deepset.ai/v2.0/docs/data-classes#chatmessage

File: integrations/mistral/src/haystack_integrations/components/generators/mistral/chat/chat_generator.py [hunk 2]
```diff
@@ -20,9 +20,21 @@ class MistralChatGenerator(OpenAIChatGenerator):
...
+    Note that the Mistral API does not accept `system` messages yet. You can use `user` and `assistant` messages.
```
Comment (anakin87) [inline integrations/mistral/src/haystack_integrations/components/generators/mistral/chat/chat_generator.py:32]:
```suggestion
```
This changed recently: https://docs.mistral.ai/api/

File: integrations/mistral/src/haystack_integrations/components/generators/mistral/chat/chat_generator.py [hunk 3]
```diff
@@ -65,15 +65,19 @@ def __init__(
...
+        :param streaming_callback:
+            A callback function that is called when a new token is received from the stream.
             The callback function accepts StreamingChunk as an argument.
-        :param api_base_url: The Mistral API Base url, defaults to `https://api.mistral.ai/v1`.
-                             For more details, see Mistral [docs](https://docs.mistral.ai/api/).
-        :param generation_kwargs: Other parameters to use for the model. These parameters are all sent directly to
-            the Mistrak endpoint. See [Mistral API docs](https://docs.mistral.ai/api/t) for
-            more details.
+        :param api_base_url:
+            The Mistral API Base url, defaults to `https://api.mistral.ai/v1`.
```
Comment (anakin87) [inline integrations/mistral/src/haystack_integrations/components/generators/mistral/chat/chat_generator.py:76]: we can remove the default

File: integrations/mistral/src/haystack_integrations/components/generators/mistral/chat/chat_generator.py [hunk 4]
```diff
@@ -65,15 +65,19 @@ def __init__(
...
+            For more details, see Mistral [docs](https://docs.mistral.ai/api/).
+        :param generation_kwargs:
+            Other parameters to use for the model. These parameters are all sent directly to
+            the Mistrak endpoint. See [Mistral API docs](https://docs.mistral.ai/api/t) for more details.
```
Comment (anakin87) [inline integrations/mistral/src/haystack_integrations/components/generators/mistral/chat/chat_generator.py:80]:
```suggestion
            the Mistral endpoint. See [Mistral API docs](https://docs.mistral.ai/api/) for more details.
```

File: integrations/mistral/src/haystack_integrations/components/generators/mistral/chat/chat_generator.py [hunk 5]
```diff
@@ -20,9 +20,21 @@ class MistralChatGenerator(OpenAIChatGenerator):
     directly to this component via the `**generation_kwargs` parameter in __init__ or the `**generation_kwargs`
     parameter in `run` method.
```
Comment (anakin87) [inline integrations/mistral/src/haystack_integrations/components/generators/mistral/chat/chat_generator.py:21]:
```suggestion
    directly to this component via the `generation_kwargs` parameter in `__init__` or the `generation_kwargs`
    parameter in `run` method.
```
These are not actual kwargs, but dicts.

**Review by anakin87 — APPROVED** [review]
I left some minor comments.

Also, these lines are outdated:
https://github.com/deepset-ai/haystack-core-integrations/blob/f758de9361272d8b65ed03a78d59e95c5d464bde/integrations/mistral/src/haystack_integrations/components/generators/mistral/chat/chat_generator.py#L15-L18

We also support mistral-large, recently released.

File: integrations/mistral/src/haystack_integrations/components/embedders/mistral/text_embedder.py [hunk 6]
```diff
@@ -38,14 +37,19 @@ def __init__(
...
-        :param prefix: A string to add to the beginning of each text.
-        :param suffix: A string to add to the end of each text.
+        Creates an MistralTextEmbedder component.
+
+        :param api_key:
+            The Mistral API key.
+        :param model:
+            The name of the Mistral embedding model to be used.
+        :param api_base_url:
+            The Mistral API Base url, defaults to `https://api.mistral.ai/v1`.
```
Comment (masci) [inline integrations/mistral/src/haystack_integrations/components/embedders/mistral/text_embedder.py:47]:
```suggestion
            The Mistral API Base url.
```

File: integrations/mistral/src/haystack_integrations/components/generators/mistral/chat/chat_generator.py [hunk 7]
```diff
@@ -17,12 +17,23 @@ class MistralChatGenerator(OpenAIChatGenerator):
...
 
+    Key Features and Compatibility:
+    - **Primary Compatibility**: Designed to work seamlessly with the Mistral API Chat Completion endpoint.
+    - **Streaming Support**: Supports streaming responses from the Mistral API Chat Completion endpoint.
+    - **Customizability**: Supports all parameters supported by the Mistral API Chat Completion endpoint.
+
+    This component uses the ChatMessage format for structuring both input and output,
+    ensuring coherent and contextually relevant responses in chat-based text generation scenarios.
+    Details on the ChatMessage format can be found in the
+    [OpenAI docs](https://github.com/openai/openai-python/blob/main/chatml.md)
```
Comment (masci) [inline integrations/mistral/src/haystack_integrations/components/generators/mistral/chat/chat_generator.py:31]:
```suggestion
    [Haystack docs](https://docs.haystack.deepset.ai/v2.0/docs/data-classes#chatmessage)
```

File: integrations/mistral/src/haystack_integrations/components/generators/mistral/chat/chat_generator.py [hunk 8]
```diff
@@ -65,15 +64,19 @@ def __init__(
...
+        :param streaming_callback:
+            A callback function that is called when a new token is received from the stream.
             The callback function accepts StreamingChunk as an argument.
-        :param api_base_url: The Mistral API Base url, defaults to `https://api.mistral.ai/v1`.
-                             For more details, see Mistral [docs](https://docs.mistral.ai/api/).
-        :param generation_kwargs: Other parameters to use for the model. These parameters are all sent directly to
-            the Mistrak endpoint. See [Mistral API docs](https://docs.mistral.ai/api/t) for
-            more details.
+        :param api_base_url:
+            The Mistral API Base url, defaults to `https://api.mistral.ai/v1`.
```
Comment (masci) [inline integrations/mistral/src/haystack_integrations/components/generators/mistral/chat/chat_generator.py:75]:
```suggestion
            The Mistral API Base url.
```

**Pull request merged**